packages/spacecat-shared-ims-client/src/clients/ims-client.js

```diff
@@ -0,0 +1,254 @@
+
+    // Always return the first product context item
+    return pcData?.productContexts?.[0];
+  }
+
+  async #getUsersByImsGroupId(imsOrgId, groupId) {
+    // This endpoint is paginated, but the default page limit is 50 entries — more than enough
+    // for our use case
+    const groupResponse = await this.#imsApiCall(
+      getGroupMembersEndpoint(imsOrgId, groupId),
+      { client_id: this.config.clientId },
+    );
+
+    if (!groupResponse.ok) {
+      throw new Error(`IMS getUsersByImsGroupId request failed with status: ${groupResponse.status}`);
+    }
+
+    const group = await groupResponse.json();
+
+    return group.items || [];
+  }
+
+  async #getUsersInAdminGroup(imsOrgId, groups) {
+    if (!Array.isArray(groups)) {
+      return [];
+    }
+
+    // Store users by their email address initially to de-dupe the entries
+    const users = {};
+    for (const group of groups) {
+      // Only process Administrators groups
+      if (group?.role === 'GRP_ADMIN') {
+        // eslint-disable-next-line no-await-in-loop
+        const groupUsers = await this.#getUsersByImsGroupId(imsOrgId, group?.ident);
+        for (const user of groupUsers) {
+          // Fallback to username if email is not set
+          const newUser = { ...user };
+          if (!hasText(newUser.email)) {
+            newUser.email = newUser.username;
+          }
+          if (emailAddressIsAllowed(newUser.email)) {
+            // Reduce fields in user object to those we need
+            users[newUser.email] = {
+              email: newUser.email,
+              firstName: newUser.firstName,
+              lastName: newUser.lastName,
+            };
+          }
+        }
+      }
+    }
+
+    // Transform the object "map" back to a de-duped array
+    return Object.keys(users).map((email) => users[email]);
+  }
+
+  async getServiceAccessToken() {
+    if (hasText(this.serviceAccessToken?.access_token)) {
+      return this.serviceAccessToken;
+    }
+
+    try {
+      const startTime = process.hrtime.bigint();
+
+      const tokenResponse = await httpFetch(
+        `https://${this.config.imsHost}${IMS_TOKEN_ENDPOINT}`,
+        {
+          method: 'POST',
+          body: createFormData({
+            client_id: this.config.clientId,
+            client_secret: this.config.clientSecret,
+            code: this.config.clientCode,
+            grant_type: 'authorization_code',
+          }),
+        },
+      );
+
+      this.#logDuration('IMS getServiceAccessToken request', startTime);
+
+      if (!tokenResponse.ok) {
+        throw new Error(`IMS getServiceAccessToken request failed with status: ${tokenResponse.status}`);
+      }
+
+      /* eslint-disable camelcase */
+      const { access_token, token_type, expires_in } = await tokenResponse.json();
+
+      this.serviceAccessToken = {
+        access_token,
+        expires_in,
+        token_type,
+      };
+
+      return this.serviceAccessToken;
+    } catch (error) {
```

Comment:
Is this catch block necessary? An error fetching a token would be thrown above (line 207) then re-thrown in the catch.